### PR TITLE
chore(localdebug): update provision and start services for v3

### PR DESCRIPTION
**Pull request by XiaofuHuang**

### CODE REVIEW

File: templates/scenarios/sso-tab-js/teamsfx/app.local.yml [hunk 1]
```diff
@@ -1,18 +1,58 @@
 version: 1.0.0
 
-# provision:
+registerApp:
+  - uses: env/generate
```
Comment (kuojianlu) [inline templates/scenarios/sso-tab-js/teamsfx/app.local.yml:4]:
```suggestion
  - uses: env/generate # Generate env to .env file
```

Comment (kuojianlu) [inline templates/scenarios/sso-tab-js/teamsfx/app.local.yml:4]: I think this driver should be placed to `configureApp`.

Comment (XiaofuHuang) [reply templates/scenarios/sso-tab-js/teamsfx/app.local.yml]: Thx, updated

Comment (XiaofuHuang) [reply templates/scenarios/sso-tab-js/teamsfx/app.local.yml]: Removed it.

File: templates/scenarios/sso-tab-js/teamsfx/app.local.yml [hunk 2]
```diff
@@ -1,18 +1,58 @@
...
+    with:
+      manifestTemplatePath: ./appPackage/manifest.template.json # Path to manifest
+    # Output: following environment variable will be persisted in current environment's .env file.
+    # TEAMS_APP_ID: the id of Teams app
+
+configureApp:
+  - uses: env/generate
+    with:
+      envs:
+        TAB_DOMAIN: localhost # Generate env to .env file
```
Comment (kuojianlu) [inline templates/scenarios/sso-tab-js/teamsfx/app.local.yml:26]: Comment in wrong place.

File: templates/scenarios/sso-tab-js/teamsfx/app.local.yml [hunk 3]
```diff
@@ -20,10 +20,10 @@ registerApp:
     # TEAMS_APP_ID: the id of Teams app
 
 configureApp:
-  - uses: env/generate
+  - uses: env/generate # Generate env to .env file
     with:
       envs:
-        TAB_DOMAIN: localhost # Generate env to .env file
+        TAB_DOMAIN: localhost
```
Comment (kuojianlu) [inline templates/scenarios/sso-tab-js/teamsfx/app.local.yml:26]: Add port.

**Review by xzf0587 — DISMISSED** [review]

File: templates/scenarios/sso-tab-js/teamsfx/app.local.yml (hunk 3 above)
Comment (XiaofuHuang) [reply templates/scenarios/sso-tab-js/teamsfx/app.local.yml:26]: It is used in aad and do not need port.

File: templates/scenarios/sso-tab-js/teamsfx/app.local.yml [hunk 4]
```diff
@@ -1,18 +1,58 @@
...
+    # Output: following environment variable will be persisted in current environment's .env file.
+    # TEAMS_APP_ID: the id of Teams app
 
 deploy:
   - uses: npm/command # Run npm command
     with:
       workingDirectory: .
       args: install --no-audit
-  
   - uses: tools/install # Install dependencies
```
Comment (swatDong) [inline templates/scenarios/sso-tab-js/teamsfx/app.local.yml:52]: Move tools/install as the 1st deploy step?

File: templates/scenarios/sso-tab-js/.vscode/tasks.json [hunk 5]
```diff
@@ -53,7 +53,7 @@
         {
             "label": "Start frontend",
             "type": "shell",
-            "command": "npm run dev:teamsfx",
+            "command": "node teamsfx/script/run.js teamsfx/.env.local",
```
Comment (swatDong) [inline templates/scenarios/sso-tab-js/.vscode/tasks.json:56]: Add `cwd` as well.

File: templates/scenarios/sso-tab-js/teamsfx/app.local.yml (hunk 4 above)
Comment (XiaofuHuang) [reply templates/scenarios/sso-tab-js/teamsfx/app.local.yml:48]: Got it.

File: templates/scenarios/sso-tab-js/teamsfx/app.local.yml (hunk 3 above)
Comment (swatDong) [reply templates/scenarios/sso-tab-js/teamsfx/app.local.yml]: Due to https://github.com/OfficeDev/TeamsFx/pull/6417?

File: templates/scenarios/sso-tab-js/package.json [hunk 6]
```diff
@@ -19,10 +19,10 @@
     },
     "devDependencies": {
         "cross-env": "^7.0.3",
-        "env-cmd": "^10.1.0"
+        "dotenv": "^16.0.3"
     },
     "scripts": {
-        "dev:teamsfx": "env-cmd --silent -f .env.teamsfx.local npm run start",
+        "dev": "react-scripts start",
```
Comment (swatDong) [inline templates/scenarios/sso-tab-js/package.json:25]: Can we just leverage the existing `start` script?

File: templates/scenarios/sso-tab-js/teamsfx/app.local.yml (hunk 3 above)
Comment (kuojianlu) [reply templates/scenarios/sso-tab-js/teamsfx/app.local.yml]: Yes.

**Comment by kuojianlu** [comment]
Update `preLaunchTask` in `Debug (Edge)` by the way.

File: templates/scenarios/sso-tab-js/package.json (hunk 6 above)
Comment (XiaofuHuang) [reply templates/scenarios/sso-tab-js/package.json]: Updated the script.

**Review by kuojianlu — APPROVED** [review]

**Review by swatDong — APPROVED** [review]

**Pull request merged**